learn/parse_http.go

```diff
@@ -326,66 +328,74 @@ func parseBody(contentType string, bodyStream io.Reader, statusCode int) (*pb.Da
 		bodyData = parseElem(string(body), interpret)
 	}
 
-	switch mediaType {
-	case "application/json":
+	// Handle JSON media types.
+	if isMediaJson {
 		bodyData, err = parseHTTPBodyJSON(bodyStream)
```

Comment:
It's too bad we can't mix conditions and value-matching in a single switch statement.

One way to do this would be instead of the boolean isMediaJson, have a string parseAsMediaType which starts as mediaType and is changed in the JSON case.  This would handle cases like text/csv too, since they could be parsed as text/plain instead of adding another boolean to handle that.

Reply:
I changed this method to use `pb.HTTPBody_ContentType` to indicate how to parse the body.  Now the flow is (a) interpret `mediaType` to set `parseBodyDataAs`, and (b) switch on `parseBodyDataAs` to do the parsing.